src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,476 @@
+        @. tmp = v^2
+        mul!(v2_x, D1, tmp)
+
+        # h_hpb_x = D1 * (h .* eta)
+        @. tmp = h * eta
+        mul!(h_hpb_x, D1, tmp)
+
+        # H_x = D1 * H
+        mul!(H_x, D1, H)
+
+        # H2_h_x = D1 * (H^2 / h)
+        @. H_over_h = H / h
+        @. tmp = H * H_over_h
+        mul!(H2_h_x, D1, tmp)
+
+        # w_x = D1 * w
+        mul!(w_x, D1, w)
+
+        # hvw_x = D1 * (h * v * w)
+        @. tmp = h * v * w
+        mul!(hvw_x, D1, tmp)
+
+        # Plain: h_t + (h v)_x = 0
+        #
+        # Split form for energy conservation:
+        # h_t + h_x v + h v_x = 0
+        @. dh = -(h_x * v + h * v_x)
+
+        # Plain: h v_t + h v v_x + g (h + b) h_x
+        #              + ... = 0
+        #
+        # Split form for energy conservation:
+        # h v_t + g (h (h + b))_x - g (h + b) h_x
+        #       + 1/2 h (v^2)_x - 1/2 v^2 h_x  + 1/2 v (h v)_x - 1/2 h v v_x
+        #       + λ/6 H^2 / h^2 h_x + λ/3 H_x - λ/3 H/h H_x - λ/6 (H^2 / h)_x
+        #       + λ/2 b_x - λ/2 H/h b_x = 0
+        lambda_6 = lambda / 6
+        lambda_3 = lambda / 3
+        lambda_2 = lambda / 2
+        if bathymetry_type isa BathymetryFlat
+            @. dv = -(g * h_hpb_x - g * (h + b) * h_x
+                      +
+                      0.5 * h * v2_x - 0.5 * v^2 * h_x
+                      +
+                      0.5 * hv_x * v - 0.5 * h * v * v_x
+                      + lambda_6 * (H_over_h * H_over_h * h_x - H2_h_x)
+                      + lambda_3 * (1 - H_over_h) * H_x) / h
+        else
+            @. dv = -(g * h_hpb_x - g * (h + b) * h_x
+                      +
+                      0.5 * h * v2_x - 0.5 * v^2 * h_x
+                      +
+                      0.5 * hv_x * v - 0.5 * h * v * v_x
+                      + lambda_6 * (H_over_h * H_over_h * h_x - H2_h_x)
+                      + lambda_3 * (1 - H_over_h) * H_x
+                      + lambda_2 * (1 - H_over_h) * b_x) / h
+        end
+
+        # Plain: h w_t + h v w_x = λ - λ H / h
+        #
+        # Split form for energy conservation:
+        # h w_t + 1/2 (h v w)_x + 1/2 h v w_x
+        #       - 1/2 h_x v w - 1/2 h w v_x = λ - λ H / h
+        @. dw = (-(0.5 * hvw_x
+                   +
+                   0.5 * h * v * w_x
+                   -
+                   0.5 * h_x * v * w
+                   -
+                   0.5 * h * w * v_x) + lambda * (1 - H_over_h)) / h
+
+        # No special split form for energy conservation required:
+        # H_t + v H_x + 3/2 v b_x = w
+        if bathymetry_type isa BathymetryFlat
+            @. dH = -v * H_x + w
+        else
+            @. dH = -v * H_x - 1.5 * v * b_x + w
+        end
+    end
+
+    @trixi_timeit timer() "source terms" calc_sources!(dq, q, t, source_terms, equations,
+                                                       solver)
+
+    return nothing
+end
+
+@inline function prim2cons(q, equations::HyperbolicSerreGreenNaghdiEquations1D)
```

Comment:
Would it maybe make sense to add another conversion function, which just returns the first three variables of the primitive variables, something like `prim2physical` (or similar)? I imagine sometimes one might be interested only in these and then one could use the conversion, e.g. when plotting by passing `conversion_function = prim2physical` to `plot`. For the other equations this function would then simply be 
```julia
prim2physical(q, equations::AbstractShallowWaterEquations) = prim2prim(q, equations)
```

Reply:
Should we make `prim2physical` the default for plotting?

Reply:
Yes sounds good.